rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,461 @@
+- No support for grid props since it isn't supported natively.
+- The TypeScript typings and autosuggestions should appropriately be adjusted for web and native to accommodate the differences, eg. grid properties, `inline-flex`, `block` display properties won't be supported. This can be done with [`moduleSuffixes`](https://www.typescriptlang.org/tsconfig#moduleSuffixes) detailed in the [related RFC](./writing-cross-platform-typescript.md).
+
+We have an [internal Box component](../packages/blade/src/components/Box/) but it exposes lot more properties than the ones associated with layout. It seems like a good idea to have a different public `Box` component with the restrictive but fully covered layout properties from above table.
+
+#### `Flex` component
+
+A helper component over the `Box` component with `display` set as `Flex`. Arguably, this component seems unneeded without flex specific shorthand properties because the behavior can be achieved with a `display="flex"` prop on `Box` but it improves code readability. Supports same properties.
+
+#### `Grid` component
+
+A helper **web only** component over the `Box` component with `display` set as `grid`. Similar to `Flex` component, it may be argued this component is unneeded with `display="grid"` set on `Box` but it improves code readability. Supports same properties.
+
+#### `Spacer` component
+
+A helper component over the `Box` component primarily aimed at putting space in between components. Useful when the spacing is different between each sibling and can't be solved by gap based properties. `Spacer` can also be helpful for spacing in between custom components which might not support a `sx` prop. Sample usage:
+
+```tsx
+<Box
+  display="flex"
+  flexDirection="column"
+  alignItems="center"
+  marginTop="spacing.1"
+  marginBottom="spacing.2"
+  paddingX="spacing.1"
+>
+  <Button>Button 1</Button>
+  <Spacer size="spacing.1" />
+  <Button>Button 2</Button>
+  <Spacer size="spacing.2" />
+  <Button>Button 3</Button>
+</Box>
+```
+
+Supported properties:
+
+| Prop     | Description                        |
+| -------- | ---------------------------------- |
+| `size`   | Sets both height and width equally |
+| `height` | Sets height                        |
+| `width`  | Sets width                         |
+
+All properties are optional and accepts only spacing tokens. If no prop is provided the spacer is set to `flex: 1`, useful for building layouts where you might want a `Spacer` to fill all available space in between siblings.
+
+#### Responsive styles
+
+In order to ease up building responsive UIs all the style prop values will also support an alternate **object based format** that accepts breakpoints specific values, keeping mobile-first approach (`@media(min-width)`). Sample usage:
+
+```tsx
+<Box
+  display="flex"
+  flexDirection={{ base: 'column', l: 'row' }}
```

Comment:
Should we avoid using shorthands here for `l`, `m`?

Reply:
These replicate our breakpoints tokens so I feel we should use the same convention.